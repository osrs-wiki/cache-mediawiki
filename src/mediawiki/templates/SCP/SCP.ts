import { MediaWikiTemplate } from "@osrs-wiki/mediawiki-builder";

import { SCPProps } from "./SCP.types";

const SCP = (props: SCPProps): MediaWikiTemplate => {
  const params: Record<string, string | number | boolean | undefined> = {
    "1": props.skill,
    "2": props.level !== undefined ? props.level : props.experience,
    "link": props.link !== undefined ? (props.link ? "yes" : "no") : undefined,
    "name": props.name || undefined,
    "txt": props.txt || undefined,
  };

  return new MediaWikiTemplate("SCP", params);
};

export default SCP;
