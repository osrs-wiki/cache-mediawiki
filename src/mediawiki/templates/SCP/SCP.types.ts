export type SkillName =
  | "Attack"
  | "Strength"
  | "Defence"
  | "Ranged"
  | "Prayer"
  | "Magic"
  | "Runecraft"
  | "Construction"
  | "Hitpoints"
  | "Agility"
  | "Herblore"
  | "Thieving"
  | "Crafting"
  | "Fletching"
  | "Slayer"
  | "Hunter"
  | "Mining"
  | "Smithing"
  | "Fishing"
  | "Cooking"
  | "Firemaking"
  | "Woodcutting"
  | "Farming";

export type NonSkillClickpic =
  | "overall"
  | "diary"
  | "combatachievement"
  | "combat"
  | "combatstyle"
  | "dd"
  | "favour"
  | "minigame"
  | "music"
  | "raid"
  | "quest"
  | "time"
  | "total";

export type SCPSkill = SkillName | NonSkillClickpic;

export interface SCPProps {
  skill: SCPSkill;
  level?: number | string;
  link?: boolean;
  experience?: number | string;
  name?: string;
  txt?: string;
}
