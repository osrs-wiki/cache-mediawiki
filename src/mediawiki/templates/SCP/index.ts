export * from "./SCP.types";
export { default } from "./SCP";
