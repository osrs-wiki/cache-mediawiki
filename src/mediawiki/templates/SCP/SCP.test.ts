import SCP from "./SCP";

describe("SCP Template", () => {
  it("should render basic SCP with skill and level", () => {
    const scp = SCP({
      skill: "Agility",
      level: 61,
      link: true,
    });

    expect(scp).toMatchSnapshot();
  });

  it("should render SCP with experience instead of level", () => {
    const scp = SCP({
      skill: "Mining",
      experience: 40000,
      link: true,
    });

    expect(scp).toMatchSnapshot();
  });

  it("should render SCP with non-skill clickpic", () => {
    const scp = SCP({
      skill: "combat",
      level: 85,
      link: true,
    });

    expect(scp).toMatchSnapshot();
  });

  it("should handle optional parameters", () => {
    const scp = SCP({
      skill: "Thieving",
      level: 62,
      link: false,
      name: "Custom Name",
      txt: "Custom Text",
    });

    expect(scp).toMatchSnapshot();
  });

  it("should work without optional parameters", () => {
    const scp = SCP({
      skill: "Strength",
      level: 58,
    });

    expect(scp).toMatchSnapshot();
  });
});
