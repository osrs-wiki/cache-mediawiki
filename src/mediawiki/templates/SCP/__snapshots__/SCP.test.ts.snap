// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SCP Template should handle optional parameters 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "SCP",
  "options": Object {
    "1": "Thieving",
    "2": 62,
    "link": "no",
    "name": "Custom Name",
    "txt": "Custom Text",
  },
  "params": Array [],
}
`;

exports[`SCP Template should render SCP with experience instead of level 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "SCP",
  "options": Object {
    "1": "Mining",
    "2": 40000,
    "link": "yes",
  },
  "params": Array [],
}
`;

exports[`SCP Template should render SCP with non-skill clickpic 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "SCP",
  "options": Object {
    "1": "Combat",
    "2": 85,
    "link": "yes",
  },
  "params": Array [],
}
`;

exports[`SCP Template should render basic SCP with skill and level 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "SCP",
  "options": Object {
    "1": "Agility",
    "2": 61,
    "link": "yes",
  },
  "params": Array [],
}
`;

exports[`SCP Template should work without optional parameters 1`] = `
MediaWikiTemplate {
  "children": undefined,
  "name": "SCP",
  "options": Object {
    "1": "Strength",
    "2": 58,
  },
  "params": Array [],
}
`;
